### ISSUE TITLE
Single call for getting pixel value

### ISSUE THREAD

**Issue by arjunrajlab** [Kitware/UPennContrast#503]
We saturate the browser queue via pixel requests. These are fired one per layer debounced based on time. We could make a single-call endpoint and debounce them based on promise status, not time.

**Comment by bruyeret** [comment]
Debouncing based on time avoids sending requests until you stop moving your mouse
Requests are only sent when the mouse stops for long enough
But I agree that a single request for all pixels would be a lot better
@manthey should we add this endpoint to the large image plugin or should I create a custom endpoint?

**Comment by arjunrajlab** [comment]
True, yes, the debouncing is nice… but it does feel a little bit "laggy" with the current debouncing. I'm wondering if we could speed that up a bit if we just had a single call.

**Comment by bruyeret** [comment]
What do you mean by laggy? It is almost instantaneous on my computer, when I stop moving the mouse
Maybe you have a dataset with many many channels?
I don't think that a single call will speed things up, but it is a good thing to avoid making many requests at once

**Comment by arjunrajlab** [comment]
So if you use e.g. ImageJ, it updates essentially in real time. Of course, that would be a bit much considering we are going back and forth over the internet :), but perhaps we can lower the debounce time? Actually, even better would be to live update even while the mouse is moving. But I'm not sure if that's just too much server traffic. Let me know what you think, perhaps we can talk Friday. This is also really pretty low priority.

**Comment by bruyeret** [comment]
Oh yes, for a desktop application that already has all the data locally, it is very easy to get the data
ImageJ just has to do `imageBuffer[mouse.x + width * mouse.y]` to get the pixel value
The equivalent on NimbusImage is to do a `getPixel` call to the server
We could eventually download all the raw images for the current location (when not unrolling), to do as ImageJ does, but it is very heavy and we lose all the optimizations of tiling the image

**Comment by arjunrajlab** [comment]
In discussion with @bruyeret, I think perhaps it's good to just reduce the debounce time slightly. If moving to a single call would help avoid spamming the server with a lot of calls, then that's great. But if just reducing the debounce time to 1/2 or 1/3 is still acceptable, then that would probably achieve much of the effect on its own.
